kernel/Kconfig

```diff
@@ -707,10 +707,10 @@ config TICKLESS_IDLE_THRESH
 config TICKLESS_KERNEL
 	bool "Tickless kernel"
 	default y if TICKLESS_CAPABLE
+	depends on TICKLESS_IDLE
```

Comment:
Actually this is wrong now.  The timer rewrite broke the dependency here and it's entirely possible (though kind of silly) to have TICKLESS_KERNEL but not TICKLESS_IDLE.  Clearly I missed the help text, which is what should be fixed.

In fact TICKLESS_IDLE itself is mostly useless now, it exists only because some tests want it, and because one or two remaining timer drivers interpret it specially (and those will be fixed soon).